supervisor/store/git.py

```diff
@@ -156,6 +156,19 @@ async def pull(self):
                     ft.partial(self.repo.git.reset, f"origin/{branch}", hard=True)
                 )
 
+                # Update submodules
+                await self.sys_run_in_executor(
+                    ft.partial(
+                        self.repo.git.submodule.root.update,
+                        {
+                            "recursive": False,
+                            "init": True,
+                            "force_remove": True,
+                            "force_reset": True,
+                        },
+                    )
+                )
+
```

Comment:
what is about shallow and depth?

Reply:
Well, the `submodule.root.update` does now seem to support that. With **`"recursive": False`** I suppose that it implies `"depth": 1`.

https://gitpython.readthedocs.io/en/0.3/reference.html#git.objects.submodule.root.RootModule.update

Reply:
Okay. Do it not need `**` to expand it?

Reply:
Not sure what you mean. But this suggestion update direct submodules.

Reply:
https://docs.python.org/3/library/functools.html#functools.partial . Did you test this code?

Reply:
just for the records, I'm fine with the change. I ask my self just if they would work like it is in that PR and how partial work

Reply:
it _might_ work, but we should be consistent, prefix the dict with `**`